### PR TITLE
(maint) prep for 0.114.0 release

### PR DESCRIPTION
**Pull request by e-gris**
Please add all notable changes to the "Unreleased" section of the CHANGELOG.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,8 +3,11 @@
 This changelog adheres to [Keep a CHANGELOG](http://keepachangelog.com/).
 
 ## [Unreleased]
+
+## [0.114.0] - 2024-01-24
```
Comment (isaac-jha) [inline CHANGELOG.md]: Looks good except the date. It is the 25th today.

**Review by isaac-jha — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (e-gris) [reply CHANGELOG.md]: It was the 24th when I submitted it :*)

**Comment by e-gris** [comment]
Let's wait until after the PE release to merge this.

**Review by isaac-jha — APPROVED** [review]

**Comment by e-gris** [comment]
We'll try this again elsewhere

**Pull request closed without merging**